oak_dice/src/evidence.rs

```diff
@@ -74,8 +83,35 @@ pub struct RootLayerEvidence {
 }
 
 impl RootLayerEvidence {
-    pub fn get_tee_platform(&self) -> Option<TeePlatform> {
-        TeePlatform::from_repr(self.tee_platform)
+    pub fn get_tee_platform(&self) -> Result<TeePlatform, String> {
+        TeePlatform::from_repr(self.tee_platform).ok_or("invalid TEE Platform value".to_string())
+    }
+
+    pub fn get_remote_attestation_report(&self) -> Result<&[u8], String> {
+        match self.get_tee_platform()? {
+            TeePlatform::AmdSevSnp => {
+                if self.remote_attestation_report.len() < AMD_SEV_SNP_ATTESTATION_REPORT_SIZE {
+                    Err("attestation report smaller than expected".to_string())
+                } else {
+                    Ok(&self.remote_attestation_report[..AMD_SEV_SNP_ATTESTATION_REPORT_SIZE])
+                }
+            }
+            _ => Ok(&self.remote_attestation_report),
+        }
+    }
+
+    pub fn set_remote_attestation_report(&mut self, src: &[u8]) -> Result<(), String> {
+        crate::utils::padded_copy_from_slice(&mut self.remote_attestation_report, src)?;
+        Ok(())
```

Comment:
I think you could avoid the `?` on the previous line followed by the Ok here and just return that (you may need an `.into()` somewhere though)